nvflare/app_common/widgets/streaming.py

```diff
@@ -103,10 +103,11 @@ def handle_event(self, event_type: str, fl_ctx: FLContext):
 
     def add(self, tag: str, value, data_type: AnalyticsDataType, global_step: Optional[int] = None, **kwargs):
         kwargs = kwargs if kwargs else {}
-        global_step = kwargs.get("global_step", None)
-        if global_step:
+        if global_step is not None:
             if not isinstance(global_step, int):
                 raise TypeError(f"Expect global step to be an instance of int, but got {type(global_step)}")
+            kwargs[TrackConst.GLOBAL_STEP_KEY] = global_step
+        global_step = kwargs.get(TrackConst.GLOBAL_STEP_KEY, None)
         dxo = create_analytic_dxo(tag=tag, value=value, data_type=data_type, writer=self.get_writer_name(), **kwargs)
         with self.engine.new_context() as fl_ctx:
             send_analytic_dxo(self, dxo=dxo, fl_ctx=fl_ctx, event_type=self.event_type)
```

Comment:
For this part, can you help adding unit tests to test the following:


1. to_dxo, from_dxo => make sure things are consistent
2. Adding different test cases to this "add" method, for example:
    - case 1: global_step = None, kwargs = None
    - case 2: global_step = 2, kwargs = None
    - case 3: global_step = None, kwargs = {"global_step": 3, "cool": 4}
    - case 4 (should not happen): global_step = 10, kwargs = {"global_step": 3, "extra_stuff": "hello"}

Reply:
Looked into this, and I think that to do this right, there may need to be additional infrastructure around mocking the engine to test for events firing as expected. The scope of this is a bit larger than I expected at first, so maybe we can do this in a future PR.

Reply:
@nvkevlu agree on the event firing part, that mechanism we can add later to the unit test framework.

For now, maybe we just add unit test up to the point of "create_analytic_dxo".
Just want to make sure these if-else statements including global_steps, kwargs is working as we expected.

Something like:

```

def mock_add(tag: str, value, data_type: AnalyticsDataType, global_step: Optional[int] = None, **kwargs):
    kwargs = kwargs if kwargs else {}
    if global_step is not None:
        if not isinstance(global_step, int):
            raise TypeError(f"Expect global step to be an instance of int, but got {type(global_step)}")
        kwargs[TrackConst.GLOBAL_STEP_KEY] = global_step
    global_step = kwargs.get(TrackConst.GLOBAL_STEP_KEY, None)
    dxo = create_analytic_dxo(tag=tag, value=value, data_type=data_type, writer=self.get_writer_name(), **kwargs)
    return dxo

class TestAnalyticsSender:
    def test_add(x, y, z):
        dxo = mock_add(x, y, z)
        assert dxo == expected_dxo
            
```